wallet/src/actors/worker/methods.rs

```diff
@@ -78,6 +78,16 @@ impl Worker {
             self.params.id_hash_iterations,
         );
 
+        // If wallet ID already exists, return Err
+        if self
+            .wallets
+            .infos()?
+            .into_iter()
+            .any(|wallet| wallet.id == id)
+        {
+            return Err(Error::WalletAlreadyExists(id));
+        };
```

Comment:
We can also check if a wallet with this id already exists by looking if the database contains some key. But since we already keep the wallet ids in memory, and we don't expect to have thousands of wallets, iterating is fine.